File: EfCoreReversing/Controllers/AccountController.cs
using EfCoreReversing.Database;
using EfCoreReversing.Database.Models;
using EfCoreReversing.Transfer.Accounts;
using Microsoft.AspNetCore.Mvc;

namespace EfCoreReversing.Controllers;

internal class AccountController : MyBaseController
{
    public AccountController(MyDbContext dbContext) : base(dbContext)
    {
        
    }

    [HttpGet("/{id}")]
    public IActionResult Get([FromRoute] uint id)
    {
        Account? account = DbContext.Accounts.FirstOrDefault(a => a.Id == id);
        
        if (account is null)
        {
            return BadRequest($"No access to account with id {id}");
        }

        return Ok();
    }

    [HttpPost("create/")]
    public IActionResult Create([FromBody]AccountTransfer payload)
    {
        return Ok();
    }

    [HttpPost("update/")]
    public IActionResult Update([FromBody] AccountTransfer payload)
    {
        return Ok();
    }

    [HttpDelete("delete/{id}")]
    public IActionResult Delete([FromRoute] int id)
    {
        return Ok();
    }
}

File: EfCoreReversing/Controllers/AccountController.rs
use actix_web::{delete, get, post, web, HttpResponse, Responder};
use crate::db::{MyDbContext, accounts::Account, AccountTransfer};

// internal class AccountController : MyBaseController
pub struct AccountController {
    db_context: std::sync::Arc<MyDbContext>,
}

impl AccountController {
    pub fn new(db_context: std::sync::Arc<MyDbContext>) -> Self {
        AccountController { db_context }
    }

    #[get("/{id}")]
    pub async fn get(id: web::Path<u32>, db_context: web::Data<std::sync::Arc<MyDbContext>>) -> impl Responder {
        let accounts = &db_context.accounts;
        let account = accounts.iter().find(|a| a.id == *id);

        if account.is_none() {
            return HttpResponse::BadRequest().body(format!("No access to account with id {}", id));
        }

        HttpResponse::Ok().finish()
    }

    #[post("/create/")]
    pub async fn create(payload: web::Json<AccountTransfer>) -> impl Responder {
        HttpResponse::Ok().finish()
    }

    #[post("/update/")]
    pub async fn update(payload: web::Json<AccountTransfer>) -> impl Responder {
        HttpResponse::Ok().finish()
    }

    #[delete("/delete/{id}")]
    pub async fn delete(id: web::Path<i32>) -> impl Responder {
        HttpResponse::Ok().finish()
    }
}